investigations/Possible_Instrument_View_Improvements.md

```diff
@@ -0,0 +1,48 @@
+# Investigation of Possible Instrument 2.0 Effects on Instrument View Performance.
+This document outlines areas in which the performance of the instrument view may benefit from the introduction of the new Instrument 2.0 caching layers. Tests were carried out using the Microsoft Visual Studio debugger and profiling tools which provided information on which Instrument 1.0 methods are most heavily used when interacting with the instrument view.
+
+### Parametrised Instrument Allocation
+In several scenarios within the Instrument View (loading, switching between unwrapped views and picking) a parametrised version of the instrument is created before attempting to access detectors. Every call to `ExperimentInfo::getInstrument()` results in a call to `ExperimentInfo::makepParameterizedInstrument()`, which in turn is triggered by every call to `InstrumentActor::getInstrument()` or `ComponentActor::getComponent()`. For SANS2D, there are 255 calls to the parametrised instrument constructor, for GEM there are ~130K. Similar figures were recorded for switching instrument representations (unwrapped views). This seems to be, in general terms, a design flaw in Instrument 1.0. However, Instrument 2.0 obviates the need to access parameter maps or even the creation of a parametrised instrument. It therefore follows that any access which replaces this object creation with a table lookup could drastically improve performance.
```

Comment:
Is it not also true that accessing the [Component](https://github.com/mantidproject/mantid/blob/master/Framework/Geometry/src/Instrument.cpp#L429)
or [Detector](https://github.com/mantidproject/mantid/blob/master/Framework/Geometry/src/Instrument.cpp#L526) on a parametrised instrument results in a further allocation? Did this not show up in the performance measurements?